x/ecocredit/server/marketplace/prune.go

```diff
@@ -0,0 +1,87 @@
+package marketplace
+
+import (
+	"context"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	marketplacev1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/marketplace/v1"
+	"github.com/regen-network/regen-ledger/types/math"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"time"
+)
+
+// PruneSellOrders is an EndBlock function that moves escrowed credits back into their tradable balance and deletes orders
+// that have expired.
+func (k Keeper) PruneSellOrders(ctx context.Context) error {
```

Comment:
Why is this separate from the existing `PruneOrders` that includes both buy and sell orders? And this should be for `BeginBlocker` not `EndBlocker`, correct? This is how `PruneOrders` is wired up at the moment. This is only being called in your test and we are still deleting and not properly returning escrowed credits.

Reply:
https://github.com/regen-network/regen-ledger/blob/70d0d40d3ac95d53b9420792cce168e4ccb04649/x/ecocredit/server/expiration.go#L23-L35

Reply:
> not properly returning escrowed credits.

https://github.com/regen-network/regen-ledger/blob/70d0d40d3ac95d53b9420792cce168e4ccb04649/x/ecocredit/server/marketplace/prune.go#L29 ?

Reply:
It's not wired up in `BeginBlocker` like `PruneOrders` is. We are still deleting sell orders without returning escrowed funds.

Reply:
The implemention looks good but it needs to be wired up as is `PruneOrders`.

Reply:
> Why is this separate from the existing PruneOrders

not sure im following, is this a concern with where the function is placed?

> that includes both buy 

buy orders are not stored in state at the moment, so im not sure i see the value in adding them at this time. 

> This is only being called in the added test and we are still deleting and not properly returning escrowed credits. 

well, none of these PR's are truly wired in yet. This is just impl + unit tests. I think that concern should be addressed when we tackle integration tests.

Reply:
> is this a concern with where the function is placed?

Not a concern about where the function is placed. `PruneOrders` is currently wired up in `BeginBlocker`.

> well, none of these PR's are truly wired in yet

`PruneOrders` is already wired up in `BeginBlocker`.

> buy orders are not stored in state at the moment

Correct but the pruning buy orders was already implemented. We can delete if need be or comment out.

Reply:
https://github.com/regen-network/regen-ledger/blob/70d0d40d3ac95d53b9420792cce168e4ccb04649/x/ecocredit/server/expiration.go#L23-L35

Reply:
Thanks for the clarification in discord. Sounds like we have more migrations to do for what I'm requesting.